x-pack/plugins/alerts/README.md

```diff
@@ -293,6 +295,105 @@ server.newPlatform.setup.plugins.alerts.registerType({
 });
 ```
 
+## Role Based Access-Control
+Once you have registered your AlertType, you need to grant your users privileges to use it.
+When registering a feature in Kibana you can specify multiple types of privileges which are granted to users when they're assigned certain roles.
+
+Assuming your feature introduces its own AlertTypes, you'll want to control:
+- Which roles have all/read privileges for these AlertTypes when they're inside the feature
+- Which roles have all/read privileges for these AlertTypes when they're outside the feature (in another feature or in the global alerts management)
+
+In addition, when users are inside your feature you might want to grant them access to AlertTypes from other features, such as built-in AlertTypes or AlertTypes provided by other features.
```

Comment:
Yeah, it's confusing, no worries.

It works like this:
Producer privilege throughout Kibana & Consumer privilege to create built-in types in Alerts Management are automatic.
Consumer privilege to create built-in types in a consumer that isn't Alerts Management is **not** automatic.

We did this so that:
1. Gaining `all` access to create alerts in `siem`, for example, doesn't automatically grant you the right to create any built-in with `siem` as consumer- as that might mean that if they run a `find` for all `siem` alerts they'll get back AlertTypes they don't support or expect.
2. What if `siem` do actually want to allow a built-in type, but only to a certain role or as a sub-privilege? We'd have to provide for that and doing so automatically now isn't something we can (necessarily) easily dial back, so best to keep this explicit for now (I think).

Reply:
Thanks, I'll process this soon.

It looks like @kobelb has a similar question as well: https://github.com/elastic/kibana/issues/43994#issuecomment-642929976.